### PR TITLE
Fix regr_r2 failure in AggregationFuzzer

### LINKED ISSUE #9089: Presto aggregate function regr_r2 fails in AggregationFuzzer
### Description

The regr_r2 aggregate function fails with the reference db not matching Velox results:
```
I20240314 21:00:30.197078 267558 AggregationFuzzer.cpp:325] ==============================> Started iteration 224 (seed: 1887832434)
I20240314 21:00:30.346477 267558 AggregationFuzzerBase.cpp:376] Executing query plan: 
-- Aggregation[SINGLE [g0, g1] a0 := regr_r2(ROW["c0"],ROW["c1"]) ORDER BY c0 ASC NULLS LAST, c1 ASC NULLS LAST] -> g0:VARCHAR, g1:ROW<f0:SMALLINT,f1:BIGINT,f2:INTEGER>, a0:REAL
  -- Values[1000 rows in 10 vectors] -> c0:REAL, c1:REAL, g0:VARCHAR, g1:ROW<f0:SMALLINT,f1:BIGINT,f2:INTEGER>
I20240314 21:00:30.364642 287255 Task.cpp:1092] All drivers (1) finished for task test_cursor 7698 after running for 17 ms.
I20240314 21:00:30.364692 287255 Task.cpp:1822] Terminating task test_cursor 7698 with state Finished after running for 17 ms.
/root/project/velox/exec/tests/utils/QueryAssertions.cpp:1115: Failure
Failed
Expected 721, got 721
5 extra rows, 5 missing rows
5 of extra rows:
        null | [null,null,246249320] | null
        null | [29447,null,1957916369] | null
 ...
 ```

### Error Reproduction

https://app.circleci.com/pipelines/github/facebookincubator/velox/46035/workflows/8b0cd41b-3035-443c-9c06-ce44cc46d5bf/jobs/318808

### Relevant logs

```bash
I20240314 21:00:30.197078 267558 AggregationFuzzer.cpp:325] ==============================> Started iteration 224 (seed: 1887832434)
I20240314 21:00:30.346477 267558 AggregationFuzzerBase.cpp:376] Executing query plan: 
-- Aggregation[SINGLE [g0, g1] a0 := regr_r2(ROW["c0"],ROW["c1"]) ORDER BY c0 ASC NULLS LAST, c1 ASC NULLS LAST] -> g0:VARCHAR, g1:ROW<f0:SMALLINT,f1:BIGINT,f2:INTEGER>, a0:REAL
  -- Values[1000 rows in 10 vectors] -> c0:REAL, c1:REAL, g0:VARCHAR, g1:ROW<f0:SMALLINT,f1:BIGINT,f2:INTEGER>
I20240314 21:00:30.364642 287255 Task.cpp:1092] All drivers (1) finished for task test_cursor 7698 after running for 17 ms.
I20240314 21:00:30.364692 287255 Task.cpp:1822] Terminating task test_cursor 7698 with state Finished after running for 17 ms.
/root/project/velox/exec/tests/utils/QueryAssertions.cpp:1115: Failure
Failed
Expected 721, got 721
5 extra rows, 5 missing rows
5 of extra rows:
        null | [null,null,246249320] | null
        null | [29447,null,1957916369] | null ..
```

### PR DESCRIPTION
**Pull request by 8dukongjian**
Fix regr_r2 for two cases:

1. m2X() = 0, then result is NULL
2. m2X() != 0 and m2Y() = 0, then result is 1.

Fixes #9089

### CODE REVIEW

File: velox/functions/prestosql/aggregates/CovarianceAggregates.cpp [hunk 1]
```diff
@@ -413,10 +413,13 @@ struct RegrSxyResultAccessor {
 
 struct RegrR2ResultAccessor {
   static bool hasResult(const ExtendedRegrAccumulator& accumulator) {
-    return accumulator.m2Y() != 0 && accumulator.m2X() != 0;
+    return accumulator.m2X() != 0;
```
Comment (mbasmanova) [inline velox/functions/prestosql/aggregates/CovarianceAggregates.cpp:416]: Would you update the documentation to clarify these edge cases?

File: velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp [hunk 2]
```diff
@@ -163,6 +163,23 @@ TEST_P(CovarianceAggregationTest, allSameValue) {
   testDistinctGroupBy(aggName, data);
 }
 
+TEST_P(CovarianceAggregationTest, allSameYValue) {
```
Comment (mbasmanova) [inline velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp:166]: nit: allSameYValue -> constantY

File: velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp [hunk 3]
```diff
@@ -163,6 +163,23 @@ TEST_P(CovarianceAggregationTest, allSameValue) {
...
+  vector_size_t size = 1'000;
+  auto data = makeRowVector({
+      makeFlatVector<int32_t>(size, [](auto row) { return row % 7; }),
+      makeFlatVector<float>(size, [](auto row) { return 1; }),
```
Comment (mbasmanova) [inline velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp:170]: 'row' argument in lambda is not used; linter will complain; please, comment it out

Would you confirm that this test fails without the code changes?

**Review by mbasmanova — COMMENTED** [review]
@8dukongjian Thank you for the fix. Looks good % some nits.

File: velox/functions/prestosql/aggregates/CovarianceAggregates.cpp (hunk 1 above)
Comment (8dukongjian) [reply velox/functions/prestosql/aggregates/CovarianceAggregates.cpp:416]: Done

File: velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp (hunk 2 above)
Comment (8dukongjian) [reply velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp]: Done

File: velox/docs/functions/presto/aggregate.rst [hunk 4]
```diff
@@ -447,7 +447,8 @@ Statistical Aggregate Functions
 .. function:: regr_r2(y, x) -> double
 
     Returns the coefficient of determination of the linear regression. ``y`` is the dependent
-    value. ``x`` is the independent value.
+    value. ``x`` is the independent value. If regr_sxx(y, x) is 0, result is null. If regr_syy(y, x) is 0
+    and regr_sxx(y, x) isn't 0, result is 1;
```
Comment (mbasmanova) [inline velox/docs/functions/presto/aggregate.rst:451]: typo: ; -> .

**Review by mbasmanova — APPROVED** [review]
@8dukongjian Thanks.

File: velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp (hunk 3 above)
Comment (8dukongjian) [reply velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp:170]: The makeFlatVector function requires a lambda expression with one parameter to be passed in.

Comment (mbasmanova) [reply velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp:170]: Here is what you need to do:

```
makeFlatVector<float>(size, [](auto /*row*/) { return 1; }),
```

File: velox/docs/functions/presto/aggregate.rst (hunk 4 above)
Comment (8dukongjian) [reply velox/docs/functions/presto/aggregate.rst]: Done

**Review by Yuhta — APPROVED** [review]

**Pull request closed without merging**